src/goto-programs/goto_program.cpp

```diff
@@ -668,6 +668,65 @@ bool goto_programt::instructiont::equals(const instructiont &other) const
   // clang-format on
 }
 
+bool goto_programt::instructiont::check_internal_invariants(
+  const symbol_tablet &table,
+  messaget &msg) const
+{
+  bool found_violation = false;
+  std::vector<std::string> id_collector;
+  auto symbol_finder = [&](const exprt &e) {
+    if(e.id() == ID_symbol)
```

Comment:
I think we should also check the types with IDs `ID_symbol_type`, `ID_struct_tag`, `ID_c_enum_tag`, and `ID_union_tag` here (see `std_types.h`). Those types also hold an identifier which indexes into the symbol table.

Reply:
Ok, I've added those checks and reworked the code to use `DATA_CHECK`.